core/src/labels/labels.cpp

```diff
@@ -58,15 +58,17 @@ void Labels::processLabelUpdate(const ViewState& viewState,
         ScreenTransform transform { m_transforms, transformRange };
 
         // Use extendedBounds when labels take part in collision detection.
-        auto bounds = (onlyTransitions || !label->canOcclude())
+        auto bounds = (onlyRender || !label->canOcclude())
             ? screenBounds
             : extendedBounds;
 
         if (!label->update(mvp, viewState, &bounds, transform)) {
             continue;
         }
 
-        if (onlyTransitions) {
+        if (label->forceInvisible()) { label->occlude(); }
```

Comment:
I wonder if this wouldn't start a fading event.

What about modifying the test line 53 to:
```
if (!drawAll && (label->state() == Label::State::dead || label->forceInvisible))
```
So the label wouldn't even be put in the collidable set.